src/main/java/br/com/alura/comex/RelatorioCategoria.java

```diff
@@ -0,0 +1,80 @@
+package br.com.alura.comex;
+
+import java.math.BigDecimal;
+import java.util.*;
+import java.util.stream.Collectors;
+
+public class RelatorioCategoria extends Relatorio{
+
+    public RelatorioCategoria(List<Pedido> listaDePedidos) {
+        super(listaDePedidos);
+    }
+
+    public int getTotalDeCategorias(){
+        HashSet<String> categoriasProcessadas = new HashSet<>();
```

Comment:
```suggestion
        Set<String> categoriasProcessadas = new HashSet<>();
```

É comum usarmos classes concretas apenas na instanciação de coleções. De resto, usar a interface!

Reply:
Ou poderia não ter o HashSet e usar o `distinct` da `Stream`:

```java
public long getTotalDeCategorias(){
        return this.listaDePedidos.stream()
                .map(Pedido::getCategoria)
                .distinct()
                .count();
    }
/```